flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -81,9 +83,24 @@ private static Config defaultConfig() {
    * @return a {@link Result} with the value and throwable (if thrown)
    */
   public static <T> Result<T> runTask(Task<T> task, Config config) {
+    Optional<TerminationHook> terminationHook = loadTerminationHook();
+    if (terminationHook.isPresent()) {
+      return new Result<>(new FloRunner<T>(config).run(task), terminationHook.get());
```

Comment:
Not important. How do you think having one constructor for Result and here we default to System:exit? So it would become terminationHook.getOrElse here and we don’t need branches.

Reply:
I agree